google-apis-core/lib/google/apis/core/http_command.rb

```diff
@@ -355,7 +358,46 @@ def safe_response_representation http_res
           http_res.inspect
         end
 
+        def opentelemetry_begin_span
+          return unless Google::Apis::Core::OpenTelemetry.instance.installed?
+          return if @opentelemetry_span
+
+          attributes = {
+            'http.host' => url.host.to_s,
+            'http.method' => method.to_s,
+            'http.target' => url.path.to_s,
+            'peer.service' => url.host.to_s
+          }
+
+          @opentelemetry_span = Google::Apis::Core::OpenTelemetry.instance.tracer.start_span(url.host.to_s, attributes: attributes)
```

Comment:
> In general, what's the best practice for naming spans? This just provides the service name. Should we include the method being called?
https://github.com/googleapis/google-api-ruby-client/pull/3185#discussion_r591982855

That's a good question as this isn't always exactly an HTTP span, if the consumers of this instrumentation also include the [http client instrumentation](https://github.com/open-telemetry/opentelemetry-ruby/tree/main/instrumentation/http_client) these spans will be wrapping the HTTP calls.

For our internal instrumentation we've been using `url.host.to_s` as the span name.